src/common/src/types/to_binary.rs

```diff
@@ -13,39 +13,75 @@
 // limitations under the License.
 
 use bytes::{Bytes, BytesMut};
+use chrono::{TimeZone, Utc};
 use postgres_types::{ToSql, Type};
 
-use super::{DatumRef, ScalarRefImpl};
+use super::{DataType, DatumRef, ScalarRefImpl};
 use crate::error::Result;
 // Used to convert ScalarRef to text format
 pub trait ToBinary {
+    // to_text impl to_binary_with_type using default type of ScalarRefImpl.
     fn to_binary(&self) -> Result<Option<Bytes>>;
+    fn to_binary_with_type(&self, ty: DataType) -> Result<Option<Bytes>>;
 }
 
 // implement use to_sql
 macro_rules! implement_using_to_sql {
-    ($({ $scalar_type:ty } ),*) => {
+    ($({ $scalar_type:ty , $data_type:ident} ),*) => {
         $(
             impl ToBinary for $scalar_type {
                 fn to_binary(&self) -> Result<Option<Bytes>> {
                     let mut output = BytesMut::new();
                     self.to_sql(&Type::ANY, &mut output).unwrap();
                     Ok(Some(output.freeze()))
                 }
+                fn to_binary_with_type(&self, ty: DataType) -> Result<Option<Bytes>> {
+                    match ty {
+                        DataType::$data_type => self.to_binary(),
+                        _ => unreachable!(),
+                    }
+                }
             }
         )*
     };
 }
 
 implement_using_to_sql! {
-    { i16  },
-    { i32  },
-    { i64  },
-    { &str },
-    { crate::types::OrderedF32 },
-    { crate::types::OrderedF64 },
-    { bool },
-    { &[u8] }
+    { i16,Int16  },
+    { i32,Int32  },
+    { &str,Varchar },
+    { crate::types::OrderedF32,Float32 },
+    { crate::types::OrderedF64,Float64 },
+    { bool,Boolean },
+    { &[u8],Bytea }
+}
+
+impl ToBinary for i64 {
+    fn to_binary(&self) -> Result<Option<Bytes>> {
+        let mut output = BytesMut::new();
+        self.to_sql(&Type::ANY, &mut output).unwrap();
+        Ok(Some(output.freeze()))
+    }
+
+    fn to_binary_with_type(&self, ty: DataType) -> Result<Option<Bytes>> {
+        match ty {
+            DataType::Int64 => self.to_binary(),
+            DataType::Timestampz => {
+                // Just a meaningful representation as placeholder. The real implementation depends
+                // on TimeZone from session. See #3552.
```

Comment:
tbh this only applies to text format. The binary format is agnostics of TimeZone.

Reply:
Why binary format is TimeZone-agnostic(Why it different with text format